packages/bolt-integration/src/bolt-payment-strategy.ts

```diff
@@ -0,0 +1,370 @@
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this.boltClient;
+    }
+
+    private getBoltEmbedded() {
+        if (!this.boltEmbedded) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this.boltEmbedded;
+    }
+
+    private async hasBoltAccount(): Promise<boolean> {
+        const state = this.paymentIntegrationService.getState();
+        const customer = state.getCustomer();
+        const billingAddress = state.getBillingAddress();
+        const email = customer?.email || billingAddress?.email || '';
+        const boltClient = this.getBoltClient();
+
+        try {
+            return await boltClient.hasBoltAccount(email);
+        } catch {
+            throw new PaymentMethodInvalidError();
+        }
+    }
+
+    private async setBoltOrderId() {
+        const order = this.paymentIntegrationService.getState().getOrderOrThrow();
+        const boltClient = this.getBoltClient();
+
+        try {
+            await boltClient.setOrderId(order.orderId);
+        } catch {
+            throw new PaymentMethodInvalidError();
+        }
+    }
+
+    private mountBoltEmbeddedField(containerId: string) {
+        const boltEmbedded = this.getBoltEmbedded();
+        const styles = { backgroundColor: '#fff' };
+        const embeddedField = boltEmbedded.create('payment_field', {
+            styles,
+            renderSeparateFields: true,
+        });
+
+        embeddedField.mount(`#${containerId}`);
+
+        this.embeddedField = embeddedField;
+    }
+
+    private validateTokenizeResultOrThrow(
+        tokenizeResult?: BoltEmbeddedTokenize | Error,
+    ): BoltEmbeddedTokenize {
+        if (!tokenizeResult) {
+            throw new PaymentMethodInvalidError();
+        }
+
+        if (tokenizeResult instanceof Error) {
+            throw new BoltError(tokenizeResult.message);
+        }
+
+        const { token, last4, bin, expiration } = tokenizeResult;
+        const lastFourDigits = +last4;
+        const iin = +bin;
+        const expirationMonth = +`${expiration}`.split('-')[1];
+        const expirationYear = +`${expiration}`.split('-')[0];
+
+        if (
+            !token ||
+            Number.isNaN(lastFourDigits) ||
+            Number.isNaN(iin) ||
+            Number.isNaN(expirationMonth) ||
+            Number.isNaN(expirationYear)
+        ) {
+            throw new PaymentArgumentInvalidError();
+        }
+
+        return tokenizeResult;
+    }
+
+    private setExtraItemsForAnalytics() {
+        const state = this.paymentIntegrationService.getState();
+        const storeConfig = state.getStoreConfigOrThrow();
+        const checkout = state.getCheckoutOrThrow();
+
+        if (storeConfig.checkoutSettings.isAnalyticsEnabled && isAnalyticsTrackerWindow(window)) {
+            const {
+                cart: { id, lineItems },
+            } = checkout;
```

Comment:
Are there any difference between `state.getCheckoutOrThrow().cart` and `state.getCartOrThrow()`?

Reply:
looks like no difference, replaced.